packages/ra-ui-materialui/src/input/DateInput.tsx

```diff
@@ -213,10 +210,11 @@ const convertDateToString = (value: Date) => {
     const yyyy = value.getFullYear().toString();
     const MM = (value.getMonth() + 1).toString();
     const dd = value.getDate().toString();
-    return `${yyyy}-${(pad + MM).slice(-2)}-${(pad + dd).slice(-2)}`;
+    const result = `${yyyy}-${(pad + MM).slice(-2)}-${(pad + dd).slice(-2)}`;
+    return result;
 };
 
-const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
+const dateRegex = /^(\d{4}-\d{2}-\d{2}).*$/;
```

Comment:
**nitpick (non-blocking):** You could probably optimize this regex by using `/^\d{4}-\d{2}-\d{2}/` instead of `/^(\d{4}-\d{2}-\d{2}).*$/` (notice the removal of `$`).
Just a nitpick though, feel free to ignore.

Reply:
Optimize?

Reply:
`.*` is a greedy operator, which can be avoided since we only actually care about the start of the string.
But anyways, on such small strings it would have almost no effect, so let's not bother with this.